From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 18 Sep 2017 00:37:17 +0100
Subject: xfrm: Avoid ABI change in 3.2.93
Forwarded: not-needed

Commit 9b3eb54106cf6acd03f07cf0ab01c13676a226c2 "xfrm: fix stack
access out of bounds with CONFIG_XFRM_SUB_POLICY" removed two members
from xfrm_dst that were written and never read.  We should retain them
as always-zero to avoid ABI breakage.

---
--- a/include/net/xfrm.h
+++ b/include/net/xfrm.h
@@ -922,6 +922,11 @@ struct xfrm_dst {
 	struct flow_cache_object flo;
 	struct xfrm_policy *pols[XFRM_POLICY_TYPE_MAX];
 	int num_pols, num_xfrms;
+#ifdef CONFIG_XFRM_SUB_POLICY
+	/* bwh: unused, for binary compatibility */
+	struct flowi *origin;
+	struct xfrm_selector *partner;
+#endif
 	u32 xfrm_genid;
 	u32 policy_genid;
 	u32 route_mtu_cached;
